mlflow/utils/rest_utils.py

```diff
@@ -8,56 +8,77 @@
 from mlflow.utils.logging_utils import eprint
 
 
-def get_databricks_hostname_and_auth():
+def _fail_malformed_databricks_auth(profile):
+    raise Exception("Got malformed Databricks CLI profile '%s'. Please make sure the Databricks "
+                    "CLI is properly configured as described at "
+                    "https://github.com/databricks/databricks-cli." % profile)
+
+
+def get_databricks_http_request_params_or_fail(profile=None):
     """
-    Reads the hostname & auth token to use for running on Databricks from the config file created
-    by the Databricks CLI. Returns a tuple of (hostname, auth, token) to use when making API
-    requests.
+    Reads in configuration necessary to make HTTP requests to a Databricks server. This
+    uses the Databricks CLI's ConfigProvider interface to load the DatabricksConfig object.
+    This method will throw an exception if sufficient auth cannot be found.
+
+    :param profile: Databricks CLI profile. If not provided, we will read the default profile.
+    :return: Dictionary with parameters that can be passed to http_request(). This will
+             at least include the hostname and headers sufficient to authenticate to Databricks.
     """
-    profile = provider.DEFAULT_SECTION
+    if not profile:
+        profile = provider.DEFAULT_SECTION
     config = provider.get_config_for_profile(profile)
+
+    hostname = config.host
+    if not hostname:
+        _fail_malformed_databricks_auth(profile)
+
+    basic_auth_str = None
     if config.username is not None and config.password is not None:
-        return config.host, (config.username, config.password), config.token
-    if config.token:
-        return config.host, None, config.token
-    raise Exception("Got malformed Databricks CLI profile '%s'. Please make sure the Databricks "
-                    "CLI is properly configured as described at "
-                    "https://github.com/databricks/databricks-cli." % profile)
+        basic_auth_str = ("%s:%s" % (config.username, config.password)).encode("utf-8")
+    elif config.token:
```

Comment:
Minor behavior change: we now defer to username/password if set. I can flip the order here if anyone cares, but they should be mostly disjoint.